### PR TITLE
Alert Manager enhancement changes

### PR DESCRIPTION
**Pull request by bibinss**
- moved existing am code to deprecate-alertmanager module. This can be deleted later
- created api module which hosts all the apis. For now only api definitions/skeletons are added
- model classes goes into 'commons' module

### CODE REVIEW

File: commons/src/main/java/com/expedia/alertmanager/model/BaseSubscription.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.expedia.alertmanager.model;
+
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+public abstract class BaseSubscription {
+    private ExpressionTree expression;
```
Comment (bibinss) [inline commons/src/main/java/com/expedia/alertmanager/model/BaseSubscription.java:9]: reference - https://github.com/ExpediaDotCom/haystack-idl/blob/master/proto/api/traceReader.proto#L62

File: commons/src/main/java/com/expedia/alertmanager/model/Dispatcher.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.expedia.alertmanager.model;
+
+import lombok.Data;
+
+@Data
+public class Dispatcher {
```
Comment (bibinss) [inline commons/src/main/java/com/expedia/alertmanager/model/Dispatcher.java:6]: this class holds the dispatcher details - EMAIL, SLACK ... etc

File: api/src/main/java/com/expedia/alertmanager/api/web/SubscriptionController.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+@RestController
+public class SubscriptionController {
+
+    @RequestMapping(value = "/api/subscriptions", method = RequestMethod.POST)
+    public List<SubscriptionResponse> createSubscriptions(@RequestBody List<CreateSubscriptionRequest> createSubRqs) {
```
Comment (bibinss) [inline api/src/main/java/com/expedia/alertmanager/api/web/SubscriptionController.java:24]: create subscription API

File: api/src/main/java/com/expedia/alertmanager/api/web/SubscriptionController.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    private void validateDispatcher(List<Dispatcher> dispatchers) {
+        Assert.notEmpty(dispatchers, "subscription dispatchers can't empty");
+    }
+
+    private void validateExpression(ExpressionTree expression) {
+        Assert.notNull(expression, "subscription expression can't null");
+    }
+
+    @RequestMapping(value = "/api/subscriptions", method = RequestMethod.GET)
+    public List<SubscriptionResponse> getSubscription(@RequestParam(required = false) String owner,
```
Comment (bibinss) [inline api/src/main/java/com/expedia/alertmanager/api/web/SubscriptionController.java:50]: get subscription API - supports list of labels as input parameter

**Review by williewheeler — APPROVED** [review]

**Pull request merged**